tests/forms_tests/tests/test_forms.py

```diff
@@ -2113,14 +2113,22 @@ class DateTimeForm(Form):
         self.assertEqual(unbound['ti_without_microsec'].value(), now_no_ms)
 
     def test_datetime_clean_initial_callable_disabled(self):
-        now = datetime.datetime(2006, 10, 25, 14, 30, 45, 123456)
+        n = 0
+
+        def fake_now():
+            nonlocal n
+            n += 1
+            return datetime.datetime(2006, 10, 25, 14, 30, 45 + n, 123456)
 
         class DateTimeForm(forms.Form):
-            dt = DateTimeField(initial=lambda: now, disabled=True)
+            dt = DateTimeField(initial=fake_now, disabled=True)
 
         form = DateTimeForm({})
         self.assertEqual(form.errors, {})
-        self.assertEqual(form.cleaned_data, {'dt': now})
+        cleaned_value = form.cleaned_data['dt']
+        bf = form['dt']
+        self.assertEqual(bf.initial, cleaned_value)
+        self.assertEqual(cleaned_value, datetime.datetime(2006, 10, 25, 14, 30, 46))
```

Comment:
This test is super clever. I feel there's a comment or an assertion missing... 

Failure at 91c5b0885322ed808d8a455720bd83fd70f6c3a5 is this: 

```
AssertionError: datetime.datetime(2006, 10, 25, 14, 30, 47) != datetime.datetime(2006, 10, 25, 14, 30, 46, 123456)
```

Two things going on: one is that `46 != 47`, the other is that we didn't trim the microseconds. 
Both because we didn't go via `BoundField.initial`. 

As it stands it's hard to look at this test and see that that's what's being checked. 
I guess a comment before the final assert would address that. 🤔

Reply:
Very perceptive comment! I can just turn these into separate tests that are named appropriately. That would be best all-around I think for both commenting and parallel versus sequential failing.

Reply:
Super. That suits. Thanks @cjerdonek 👍